api/src/main/java/org/opennms/integration/api/v1/timeseries/TimeSeriesStorage.java

```diff
@@ -51,9 +54,106 @@ public interface TimeSeriesStorage {
      */
     List<Metric> findMetrics(Collection<TagMatcher> tagMatchers) throws StorageException;
 
-    /** Returns the data for the given metrics for the given time period. */
+    /** Returns the data for the given metrics for the given time period.
+     *  Deprecated: use getTimeSeriesData() instead.
+     *  This method will only be called if getTimeSeriesData is not implemented yet.
+     */
+    @Deprecated
     List<Sample> getTimeseries(TimeSeriesFetchRequest request) throws StorageException;
 
+    /** Returns the data for the given metric for the given time period. */
+    default TimeSeriesData getTimeSeriesData(TimeSeriesFetchRequest request) throws StorageException {
```

Comment:
to be discussed: Do we want to provide the default implementation for backwards compatibility?

Reply:
yes, we should remain backwards compatible

Reply:
ok